blockchain/message_pool/src/msgpool.rs

```diff
@@ -794,15 +908,29 @@ pub mod test_provider {
             Ok(Cid::default())
         }
 
-        fn state_get_actor(&self, addr: &Address, _ts: &Tipset) -> Result<ActorState, Errors> {
-            let s = self.state_sequence.get(addr);
-            let mut sequence = 0;
-            if let Some(sq) = s {
-                sequence = *sq;
+        fn get_actor_after(&self, addr: &Address, ts: &Tipset) -> Result<ActorState, Errors> {
+            let mut msgs: Vec<SignedMessage> = Vec::new();
+            for b in ts.blocks() {
+                if let Some(ms) = self.bmsgs.get(b.cid()) {
+                    for m in ms {
+                        if m.from() == addr {
+                            msgs.push(m.clone());
+                        }
+                    }
+                }
+            }
+            msgs.sort_by_key(|m| m.sequence());
+            let mut sequence: u64 = self.state_sequence.get(addr).copied().unwrap_or_default();
+            for m in msgs {
+                if m.sequence() != sequence {
+                    break;
+                }
+                sequence += 1;
```

Comment:
Another way to do this with one line is this but the issue is that it is less efficient. So this is fine for me.

let sequence = sequence + mgs.iter().take_while(|m|m.sequence==sequence).count()